src/rustup.ts

```diff
@@ -0,0 +1,2458 @@
+            "Toolchain name, such as 'stable', 'nightly', or '1.8.0'",
+          args: {
+            name: "<toolchain>",
+            description:
+              "Toolchain name, such as 'stable', 'nightly', or '1.8.0'",
+          },
+        },
+      ],
+    },
+    {
+      name: "doc",
+      description: "Open the documentation for the current toolchain",
+      options: [
+        {
+          name: "--alloc",
+          description: "The Rust core allocation and collections library",
+        },
+        {
+          name: "--book",
+          description: "The Rust Programming Language book",
+        },
+        {
+          name: "--cargo",
+          description: "The Cargo Book",
+        },
+        {
+          name: "--core",
+          description: "The Rust Core Library",
+        },
+        {
+          name: "--edition-guide",
+          description: "The Rust Edition Guide",
+        },
+        {
+          name: "--embedded-book",
+          description: "The Embedded Rust Book",
+        },
+        {
+          name: ["-h", "--help"],
+          description: "Prints help information",
+        },
+        {
+          name: "--nomicon",
+          description: "The Dark Arts of Advanced and Unsafe Rust Programming",
+        },
+        {
+          name: "--path",
+          description: "Only print the path to the documentation",
+        },
+        {
+          name: "--proc_macro",
+          description:
+            "A support library for macro authors when defining new macros",
+        },
+        {
+          name: "--reference",
+          description: "The Rust Reference",
+        },
+        {
+          name: "--rust-by-example",
+          description:
+            "A collection of runnable examples that illustrate various Rust concepts and standard libraries",
+        },
+        {
+          name: "--rustc",
+          description: "The compiler for the Rust programming language",
+        },
+        {
+          name: "--rustdoc",
+          description: "Generate documentation for Rust projects",
+        },
+        {
+          name: "--std",
+          description: "Standard library API documentation",
+        },
+        {
+          name: "--test",
+          description:
+            "Support code for rustc's built in unit-test and micro-benchmarking framework",
+        },
+        {
+          name: "--unstable-book",
+          description: "--unstable-book",
+        },
+        {
+          name: "--toolchain",
+          description:
+            "Toolchain name, such as 'stable', 'nightly', or '1.8.0'",
+          args: {
+            name: "<toolchain>",
+          },
+        },
+      ],
+      args: {
+        name: "<topic>",
+        description:
+          "Topic such as 'core', 'fn', 'usize', 'eprintln!', 'core::arch', 'alloc::format!', 'std::fs', 'std::fs::read_dir', 'std::io::Bytes', 'std::iter::Sum', 'std::io::error::Result' etc",
+        suggestions: [
```

Comment:
You can actually generate those all at run time which will always be up to date using a generator

```suggestion
        generators: {
          script: `find $(rustup docs --path | sed -e "s|index\\.html|std|") $(rustup docs --path | sed -e "s|index\.html|alloc|") $(rustup docs --path | sed -e "s|index\\.html|core|") | grep "\\.html" | sed -E -e "s|^(.*)/html/||" -e "s|\\.html||" -e "s|/|::|g" -e "s/constant\\.|trait\\.|struct\\.|macro\\.|fn\\.|keyword\\.|primitive\\.|type\\.|enum\\.|union\\.|traitalias\\.|::index$//"`,
          splitOn: "\n"
        },
```

Reply:
Whoa, didn’t know that. Thanks!